base/abn_metrics.go

```diff
@@ -0,0 +1,174 @@
+	// abnMetricProvider is prefix for abn metrics
+	abnMetricProvider = "abn"
+)
+
+// ABNMetricsInputs is the inputs for for the abnmetrics task
+type ABNMetricsInputs struct {
+	Endpoint *string `json:"endpoint,omitempty" yaml:"endpoint,omitempty"`
+	// Application is name of application to evaluate
+	Application string `json:"application" yaml:"application"`
+}
+
+// collectABNMetricsTask is task definition for abnmetrics task
+type collectABNMetricsTask struct {
+	TaskMeta
+	With   ABNMetricsInputs `json:"with" yaml:"with"`
+	client abnClientInterface
+}
+
+// initializeDefaults sets default values for the task
+func (t *collectABNMetricsTask) initializeDefaults() {
+	if t.client == nil {
+		ep := defaultEndpoint
+		if t.With.Endpoint != nil {
+			ep = *t.With.Endpoint
+		}
+		t.client = &defaultABNClient{
+			endpoint: ep,
+		}
+	}
+}
+
+// validate task inputs
+func (t *collectABNMetricsTask) validateInputs() error {
+	return nil
+}
+
+// run exeuctes this task
+func (t *collectABNMetricsTask) run(exp *Experiment) error {
+	var err error
+
+	// validate inputs
+	err = t.validateInputs()
+	if err != nil {
+		return err
+	}
+
+	// initialize defaults
+	t.initializeDefaults()
+
+	// get application json from abn service
+	applicationJSON, err := t.client.callGetApplicationJSON(t.With.Application)
+	if err != nil {
+		return err
+	}
+
+	// convert to Application
+	a := &abnapp.Application{}
+	err = yaml.Unmarshal([]byte(applicationJSON), a)
+	if err != nil {
+		return err
+	}
+
+	// count number of tracks
+	numTracks := len(a.Tracks)
+	if numTracks == 0 {
+		log.Logger.Warn("no tracks detected in application")
+		return nil
+	}
+
+	// initialize insights in Result with number of tracks
+	err = exp.Result.initInsightsWithNumVersions(numTracks)
+	if err != nil {
+		return err
+	}
+	log.Logger.Tracef("intialized insights with %d versions", numTracks)
+
+	in := exp.Result.Insights
+
+	// add metrics for all tracks
+	versionIndex := 0
+	in.VersionNames = make([]VersionInfo, in.NumVersions)
+	// for each track (and corresponding version), get the Version object
+	// Use it to update all metrics for this version
+	for track, version := range a.Tracks {
+		// set the track identifier/version name in result
+		in.VersionNames[versionIndex].Version = version
+		in.VersionNames[versionIndex].Track = track
+		// get version object from retrieved application object
+		v, _ := a.GetVersion(version, false)
+		if v == nil {
+			log.Logger.Debugf("expected version %s not found", version)
+			return errors.New("expected version not found")
+		}
+		log.Logger.Tracef("version %s is mapped to track %s; using index %d", version, track, versionIndex)
+		// update all metrics with new values (is summary metric so just replace)
+		for metric, m := range v.Metrics {
+			log.Logger.Tracef("   updating metric %s with data %+v", metric, m)
+			err := in.updateMetric(
+				abnMetricProvider+"/"+metric,
+				MetricMeta{
+					Description: "summary metric",
```

Comment:
The `summarymetrics.SummaryMetric` object itself should including a description (so that you get a specific description for the metric instead of a generic one).

Reply:
For now, the above looks good.

Reply:
Description is not necessary and it isn't clear that this description adds much value. I'll remove it.